Android/app/src/main/java/io/github/project_travel_mate/login/LoginActivity.java

```diff
@@ -212,4 +222,43 @@ public void showMessage(String message) {
         Toast.makeText(this, message, Toast.LENGTH_LONG)
                 .show();
     }
+
+    @Override
+    public void validateSignUp() throws ValidationException {
```

Comment:
In terms of this MVP design, I honestly feel, the validation logic (this method and 3 methods below this code) should be part of the presenter. FYI -> Views a.k.a Activities/Fragments need to be as dumb as possible and business logic should be part of your presenter.

Reply:
@momen-ghulmi Please incorporate this change also.